plan/aggregation_push_down.go

```diff
@@ -391,6 +391,16 @@ func (a *aggregationOptimizer) aggPushDown(p LogicalPlan) LogicalPlan {
 				}
 				union.SetChildren(newChildren...)
 				union.SetSchema(pushedAgg.schema)
+			} else if ds, ok1 := child.(*DataSource); ok1 {
```

Comment:
If aggregation is pushed down to storage engines, the engines should know which columns are virtual generated and how to calculate them by their generation expression respectively.
So here we rewrite the plan to replace virtual generated column. For example:
```sql
select a, max(generated_c) from t group by a 
```
will be rewrite to
```sql
select a, max(a+b) from t group by a
```